### DISCUSSION TITLE
Data Importer and GoCardless/Nordigen: How to set up auto import

### DISCUSSION THREAD

**Discussion by dekiesel** [firefly-iii/firefly-iii#8249] [Firefly III] [answered]
Hi,

I feel like I've clicked past the description a bunch of times now because people are talking about [auto import](https://docs.firefly-iii.org/data-importer/advanced/automation/) but I can't find out how to do it.

I can successfully (manually) import transactions via GoCardless, but I don't know how to tell it "get last days transactions every night at 2" for example.

If anybody has any pointers I'd be really grateful! 
Thanks!

**Comment by JC5** [comment]
There's no clear guide on this yet, I shall expand upon it. But if you get the following to work, you can simply put it into a cron job.

https://docs.firefly-iii.org/how-to/data-importer/advanced/cli/

**Reply by dekiesel** [reply]
Thanks for the link. I do have a json file and I think this section applies to me

```
docker run \
--rm \
-v $PWD:/import \
-e FIREFLY_III_ACCESS_TOKEN= \
-e IMPORT_DIR_ALLOWLIST=/import \
-e FIREFLY_III_URL= \
-e WEB_SERVER=false \
fireflyiii/data-importer:latest
```
But I don't see how I can use the generated json file here?

And wouldn't I need to log in to my bank anyway? At the moment, when loading manually, I have to input my bank login + a one time key.

**Reply by dekiesel** [reply]
@JC5 , sorry to bring this up again, but I can't get it to work without the passwords and the 2fa prompt. I'd be happy about any pointers you have!

**Comment by ShooTeX** [comment, marked as answer]
I did the following:

I have the data importer setup as described here:
https://docs.firefly-iii.org/how-to/data-importer/installation/docker/

With additions as described here:
https://docs.firefly-iii.org/how-to/data-importer/advanced/cli/

---

In the `.importer.env`:
- Add `IMPORT_DIR_ALLOWLIST=/import`
- Add value for `FIREFLY_III_ACCESS_TOKEN`
- Add `NORDIGEN` or `SPECTRE` credentials

---

in the compose file:
- bind directory to importer, in my case:
```
volumes:
  - ./import:/import
```

---

- in the Data Importer, run through the import and download the configuration file
- put the json file in the `import` folder you created and bound to the container
- run: `docker exec firefly_iii_importer php artisan importer:import /import/config.json`
  where `firefly_iii_importer` is the container id and where `config.json` is the configuration file from the Data Importer

this should now run the import and now you could put it in a cronjob:
- `crontab -e`
- add:
```
0 22 * * * docker exec firefly_iii_importer php artisan importer:import /import/config.json
```

---

I hope I didn't miss anything and this helps you

**Reply by dekiesel** [reply]
I can't believe it... This worked on the first try. THANK YOU SO MUCH!

It didn't even ask me for the bank login as it does when using the GUI. Again, thank you so much!

**Reply by JcMinarro** [reply]
Would it be possible to invoke this "PHP command" using a curl API Call??
I would like to have a "cron container" within my docker-compose instead of creating the cron task on the host machine.
The idea is to be able to do an external query to the data-importer domain to auto-import from some internal config, as you do with this docker command. Something like:
```
curl 'https://[DATA_IMPORTER_DOMAIN]/autoimport?secret=MY_SECRET' \
  --data-raw '{"importConfig": "/import/config.json"}'
```

**Reply by JC5** [reply]
Of course, just use the cron container from the [docker-compose.yml](https://github.com/firefly-iii/docker/blob/main/docker-compose-importer.yml#L50) file as an example.

You cannot push options like that however, the data importer must be pre-configured.

https://docs.firefly-iii.org/how-to/data-importer/advanced/post/

**Reply by JcMinarro** [reply]
Let say I create a couple of container:
-. DataImporter -> A
-. Cron -> B

Reading the docs you provide, the `curl` call needs to push a "local json file" config stored on "B" container, or would it be possible to store it on "A" container (enabling `JSON_CONFIGURATION_DIR` volume) but keep calling the `curl` command from "B" without push any json file config?

**Reply by JC5** [reply]
Keep reading.

https://docs.firefly-iii.org/how-to/data-importer/advanced/post/#importing-a-local-directory

**Reply by JcMinarro** [reply]
Ohh, ok, thanks, I didn't see the last point.
It will help me a lot

**Comment by JC5** [comment]
Thanks both. I'll expand the pages with some pointers.
